Assets/MRTK/Providers/LeapMotion/Editor/ConfigurationChecker/LeapMotionConfigurationChecker.cs

```diff
@@ -0,0 +1,389 @@
+
+                if (isLeapCoreAssetsVersionSupported)
+                {
+                    RemoveTestingFolders();
+                    AddAndUpdateAsmDefs();
+                    AddLeapEditorAsmDefs();
+
+                    // Refresh the database because tests were removed and 10 asmdefs were added
+                    AssetDatabase.Refresh();
+                }
+                else
+                {
+                    Debug.LogError("MRTK only supports the Leap Motion Core Assets Version 4.4.0, the Leap Motion Core Assets imported are not Version 4.4.0.");
+                }
+            }
+            
+            if (!isLeapInProject && references.Contains("LeapMotion"))
+            {
+                references.Remove("LeapMotion");
+                leapDataProviderAsmDef.References = references.ToArray();
+                leapDataProviderAsmDef.Save(leapDataProviderAsmDefFile[0].FullName);
+            }
+        }
+
+        /// <summary>
+        /// Checks if the Leap Motion Core Assets are version 4.4.0.
+        /// </summary>
+        /// <returns>True, if the Leap Motion Core Assets imported are version 4.4.0.</returns>
+        private static bool LeapCoreAssetsVersionSupport()
+        {
+            string versionLeapPath = Path.Combine(Application.dataPath, pathDifference, "LeapMotion", "Core", "Version.txt");
+
+            string leapCoreAssetsVersionNumber = "4.4.0";
+
+            using (StreamReader streamReader = new StreamReader(versionLeapPath))
+            {
+                while (streamReader.Peek() > -1)
+                {
+                    string line = streamReader.ReadLine();
+
+                    if (line.Contains(leapCoreAssetsVersionNumber))
+                    {
+                        return true;
+                    }
+                }
+
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// The Leap Core Assets currently contain multiple folders with tests in them.  An issue has been filed in the Unity
+        /// Modules repo: https://github.com/leapmotion/UnityModules/issues/1097.  The issue with the multiple test folders is when an 
+        /// asmdef is placed at the root of the core assets, each folder containing tests needs another separate asmdef.  This method
+        /// is used to avoid adding an additional 8 asmdefs to the project, by removing the folders and files that are tests in the 
+        /// Leap Core Assets.
+        /// </summary>
+        private static void RemoveTestingFolders()
+        {
+            // If one of the leap test directories exists then the rest have not been deleted
+            if (Directory.Exists(Path.Combine(Application.dataPath, pathDifference, pathsToDelete[0])))
+            {
+                foreach (string path in pathsToDelete)
+                {
+                    // What if leap is not imported to the root of assets?
```

Comment:
Nit: It would probably be good to reword this comment - i.e. I could read this as "oh maybe we don't handle leap being not in the root of assets"

Reply:
Yes I agree, the pathDifference makes sure to handle that case. The comment has been reworded.